05/dbernhard/err_function_return_type.oat

```diff
@@ -0,0 +1,26 @@
+struct A {int x}
+struct B {int x; bool y}
+struct C {int c}
+
+bool same_x(A? first, A? second) {
+  if? (A f = first) {
+    if? (A s = second) {
+      return f.x == s.x;
+    }
+  }
+  return false;
+}
+
+C get() {
+  return new C { c = 2 };
+}
+
+int program(int argc, string[] argv) {
+  if (same_x(get(), get())) {
```

Comment:
IMHO, I would say the test is more about wrong passed parameters type (C is not a subtype of A) than wrong function return type, but it is a point of view so doesn't really matter